spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ColumnarBatchReader.java

```diff
@@ -183,6 +187,8 @@ Pair<int[], Integer> buildPosDelRowIdMapping(PositionDeleteIndex deletedRowPosit
           currentRowId++;
         } else if (hasIsDeletedColumn) {
           isDeleted[originalRowId] = true;
+        } else {
+          deletes.incrementDeleteCount();
         }
```

Comment:
We still need the metrics in case of `_deleted` column, right?
```
else {
     if (hasIsDeletedColumn) {
          isDeleted[originalRowId] = true;
     }
    deletes.incrementDeleteCount();
}
```

Reply:
So, following the explanation by @RussellSpitzer: 

> there is just now a second read path with _is_deleted metadata column which actually returns deleted rows. When we follow that path I'm not sure it's important for us to count the deleted rows since we'll be returning them anyway, I could go either way. 

I decided that for this read path with the _id_deleted metadata column, we will not perform the count. That is what has been implemented here.

Pardon my ignorance: for my benefit, how does one perform such a read (where `hasIsDeletedColumn` is true)?

Reply:
It is triggered by either projection or filtering. Here are two examples:
```
select name, _deleted from student
```
```
select name from student where _deleted=true
```
It is true that user can get row-level deleted rows, users can also get both deleted and un-deleted rows with `_deleted` column, the first sql above returns that. The way I saw it is that the delete count metric is orthogonal to the metadata column `_deleted`. It reports how many row-level deletes applied as its label indicates, `public static final String DISPLAY_STRING = "number of row deletes applied"`. I don’t have strong option on it though. What do you think? BTW, the non-vectorized read also didn’t report the metric in case of `_deleted` column, see line 255 in class `DeleteFilter`.

Reply:
@flyrain thank you for explaining how reads with `hasIsDeletedColumn` occur. 
I didn't really understand that before, but based on @RussellSpitzer's comment I cited, I figured it was fine not to count deletes for such reads. I have done this consistently for both vectorized and non-vectorized reads, as you realize. This is deliberate.

Now that I understand the use case for `hasIsDeletedColumn`, I agree that there is merit to the argument for considering "number of row deletes applied" metric to apply regardless of `hasIsDeletedColumn`. On the other hand, for the queries where `hasIsDeletedColumn` is true, you are able to infer the number of row deletes applied from the result. I don't have a strong opinion on what to do in the `hasIsDeletedColumn` scenario. But the current implementation is consistent. @RussellSpitzer do you want to weigh in? 

If we want to count the deletes always, we'll need to modify `DeleteFilter#markDeleted` (we'll need to pass in the `DeleteCounter`) as well as the changes you mention for `ColumnarBatchReader`.

Reply:
I was mistaken about needing to modify `Deletes#markDeleted`, `Deletes#streamingMarker` and `Deletes.PositionStreamDeleteMarker`. I just needed to change `BaseReader.SparkDeleteFilter#markRowDeleted` to increment the counter there. `BaseReader.SparkDeleteFilter` keeps a `DeleteCounter` instance.